pkg/v5/configmap/update.go

```diff
@@ -0,0 +1,90 @@
+package configmap
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/operatorkit/controller"
+	corev1 "k8s.io/api/core/v1"
+)
+
+func (s *Service) ApplyUpdateChange(ctx context.Context, configMapConfig ConfigMapConfig, configMapsToUpdate []*corev1.ConfigMap) error {
+	if len(configMapsToUpdate) > 0 {
+		s.logger.LogCtx(ctx, "level", "debug", "message", "updating configmaps")
+
+		guestK8sClient, err := s.guest.NewK8sClient(ctx, configMapConfig.ClusterID, configMapConfig.GuestAPIDomain)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		for _, configMapToUpdate := range configMapsToUpdate {
+			_, err := guestK8sClient.CoreV1().ConfigMaps(configMapToUpdate.Namespace).Update(configMapToUpdate)
+			if err != nil {
+				return microerror.Mask(err)
+			}
+		}
+
+		s.logger.LogCtx(ctx, "level", "debug", "message", "updated configmaps")
+	} else {
+		s.logger.LogCtx(ctx, "level", "debug", "message", "no need to update configmaps")
+	}
+
+	return nil
+}
+
+func (s *Service) NewUpdatePatch(ctx context.Context, currentState, desiredState []*corev1.ConfigMap) (*controller.Patch, error) {
+	create, err := s.newCreateChange(ctx, currentState, desiredState)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	update, err := s.newUpdateChange(ctx, currentState, desiredState)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	/* TODO
+	delete, err := s.newDeleteChangeForUpdatePatch(ctx, currentState, desiredState)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+	*/
+
+	patch := controller.NewPatch()
+	patch.SetCreateChange(create)
+	patch.SetUpdateChange(update)
+	// patch.SetDeleteChange(delete)
+
+	return patch, nil
+}
+
+func (s *Service) newUpdateChange(ctx context.Context, currentConfigMaps, desiredConfigMaps []*corev1.ConfigMap) ([]*corev1.ConfigMap, error) {
+	s.logger.LogCtx(ctx, "level", "debug", "message", "finding out which configmaps have to be updated")
+
+	configMapsToUpdate := make([]*corev1.ConfigMap, 0)
+
+	for _, currentConfigMap := range currentConfigMaps {
+		desiredConfigMap, err := getConfigMapByName(desiredConfigMaps, currentConfigMap.Name)
+		if IsNotFound(err) {
+			// Ignore here. These are handled by newDeleteChangeForUpdatePatch().
+			continue
+		} else if err != nil {
+			return nil, microerror.Mask(err)
+		}
+
+		if isConfigMapModified(desiredConfigMap, currentConfigMap) {
+			// Make a copy and set the resource version so the CR can be updated.
+			configMapToUpdate := desiredConfigMap.DeepCopy()
+			configMapToUpdate.ObjectMeta.ResourceVersion = currentConfigMap.ObjectMeta.ResourceVersion
+
```

Comment:
This is the magic to set the resource version from the current state. Without this the K8s update API calls will fail.